src/ui/controls/color-picker/hsb-picker.tsx

```diff
@@ -118,15 +121,24 @@ export default function HSBPicker(props: HSBPickerProps) {
     }
 
     function onSBCanvasRender(canvas: HTMLCanvasElement) {
+        if (isElementHidden(canvas)) {
+            store.wasPrevHidden = true;
+            return;
+        }
         const hue = activeHSB.h;
         const prevHue = prevColor && rgbToHSB(parse(prevColor)).h;
-        if (hue === prevHue) {
+        if (!store.wasPrevHidden && hue === prevHue) {
```

Comment:
Maybe it will be better if you either assign a default value `true` if it is `null` or `undefined`? Or maybe it's better to rename it into `wasPrevVisible` and invert the logic everywhere? So the default value will be falsy (undefined).

Reply:
> aybe it will be better if you either assign a default value true if it is null or undefined? 

It's never null or defined if you reach this, because on start it will never be visible and then it will assign `true`

Reply:
Initially `store` object has no such property `wasPrevHidden`, so it is `undefined`.

So you could
```javascript
if (store.wasPrevHidden == null) {
  store.wasPrevHidden = true;
}
```
Oh wait, haha, I forgot how to use a library I created. There is a `context.getStore(defaults)` method, so it can be `const store = context.getStore({wasPrevHidden: true})`.

So later on this stored property will make more sense, if you understand what I mean. The initial value will be correct and this should correctly replace `&&` with `||` (if I don't miss anything again).

Reply:
Had to invert the logic and remove the `return` statement and should be good now.